src/graphics/canvas-layer.js

```diff
@@ -112,7 +115,11 @@ Crafty._registerLayerTemplate("Canvas", {
 
         // set referenced objects to initial values -- necessary to avoid shared state between systems
         this._dirtyRects = [];
+        this._dirtyCells = [];
         this._changedObjs = [];
+        this._tempKeys = {};
+        this._tempRect = { _x: 0, _y: 0, _w: 0, _h: 0 };
+        this._tempRect2 = { _x: 0, _y: 0, _w: 0, _h: 0 };
```

Comment:
Sounds good, but what's the benefit in this case exactly? The code needs only 2 rects at any time, won't it be an overhead to request and recycle them each render frame? Or would they be requested just once, like in the initialization code above? `this._tempRect = Crafty.rectManager._rectPool.get(0, 0, 0, 0)`